pytest/tests/mocknet/helpers/neard_runner.py

```diff
@@ -665,9 +669,53 @@ def do_ls_backups(self):
         with self.lock:
             return self.data.get('backups', {})
 
-    def do_update_binaries(self):
+    # Updates the URL for the given epoch height or binary idx. adds a new one if the epoch height does not exit
+    def update_binaries_url(self, neard_binary_url, epoch_height, binary_idx):
+        if neard_binary_url is not None and ((epoch_height is None)
+                                             != (binary_idx is None)):
+            logging.info(
+                f'Updating binary list for height:{epoch_height} or idx:{binary_idx} with '
+                f'url: {neard_binary_url}')
+        else:
+            logging.error(f'wrong arguments provided')
+            return
+
+        if 'binaries' not in self.config:
+            self.config['binaries'] = []
+
+        if not isinstance(self.config['binaries'], list):
+            self.config['binaries'] = []
+
+        if epoch_height is not None:
+            binary = next((b for b in self.config['binaries']
+                           if b['epoch_height'] == epoch_height), None)
+            if binary:
+                binary['url'] = neard_binary_url
+            else:
+                self.config['binaries'].append({
+                    'url': neard_binary_url,
+                    'epoch_height': epoch_height
+                })
+                self.config['binaries'].sort(
+                    key=lambda binary: binary['epoch_height'])
+        if binary_idx is not None:
+            binaries_number = len(self.config['binaries'])
+            if binary_idx >= binaries_number:
+                logging.error(
```

Comment:
again i think we should raise `raise jsonrpc.exceptions.JSONRPCDispatchException()` to give an error to the caller